src/bin/pg_autoctl/pgsql.c

```diff
@@ -1187,18 +1187,18 @@ typedef struct PgsrSyncAndWALContext
 {
 	bool		parsedOk;
 	char		syncState[PGSR_SYNC_STATE_MAXLENGTH];
-	uint64_t	walLag;
+	char        latestLSN[NAMEDATALEN];
 } PgsrSyncAndWALContext;
 
 bool
 pgsql_get_sync_state_and_wal_lag(PGSQL *pgsql, const char *slotName,
-								 char *pgsrSyncState, int64_t *walLag,
+								 char *pgsrSyncState, char *latestLSN,
 								 bool missing_ok)
 {
 	PgsrSyncAndWALContext context = { 0 };
 	char *sql =
 		"select sync_state, "
-		"pg_current_wal_lsn() - flush_lsn as wal_lag "
+		"pg_current_wal_lsn() "
```

Comment:
The same thing was named `receivedLSN` and then `latestLSN` but actually is the `current LSN` then?

Reply:
I use received_lsn at secondary node, and latest_lsn at monitor.

Reply:
having second thoughts on this.

Reply:
per our discussion, it is `current lsn` at data nodes, and `reported lsn` at monitor.